pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/AvgAggregationFunction.java

```diff
@@ -37,11 +39,16 @@ public class AvgAggregationFunction extends BaseSingleInputAggregationFunction<A
   private static final double DEFAULT_FINAL_RESULT = Double.NEGATIVE_INFINITY;
   private final boolean _nullHandlingEnabled;
 
-  public AvgAggregationFunction(ExpressionContext expression) {
-    this(expression, false);
+  public AvgAggregationFunction(List<ExpressionContext> arguments, boolean nullHandlingEnabled) {
+    this(verifyArguments(arguments), nullHandlingEnabled);
   }
 
-  public AvgAggregationFunction(ExpressionContext expression, boolean nullHandlingEnabled) {
+  private static ExpressionContext verifyArguments(List<ExpressionContext> arguments) {
```

Comment:
Shall we move `verifyArguments` to a static method in `AggregationFunctionFactor`?

Like `private static ExpressionContext verifySingleArgument(List<ExpressionContext> arguments, AggregationFunctionType funcType)`